src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -220,6 +220,293 @@ int16_t EnforceCoolingSetpointLimits(int16_t CoolingSetpoint, EndpointId endpoin
+ * @return Success if the limits were read completely, otherwise an error code
+ */
+Status GetSetpointLimits(EndpointId endpoint, SetpointLimits & setpointLimits)
+{
+    uint32_t flags;
+    if (FeatureMap::Get(endpoint, &flags) != Status::Success)
+    {
+        ChipLogError(Zcl, "GetSetpointLimits: could not get feature flags");
+        flags = FEATURE_MAP_DEFAULT;
+    }
+
+    auto featureMap                   = BitMask<Feature, uint32_t>(flags);
+    setpointLimits.autoSupported      = featureMap.Has(Feature::kAutoMode);
+    setpointLimits.heatSupported      = featureMap.Has(Feature::kHeating);
+    setpointLimits.coolSupported      = featureMap.Has(Feature::kCooling);
+    setpointLimits.occupancySupported = featureMap.Has(Feature::kOccupancy);
+
+    if (setpointLimits.autoSupported)
+    {
+        setpointLimits.deadBand = 0;
+        if (MinSetpointDeadBand::Get(endpoint, &setpointLimits.deadBand) != Status::Success)
+        {
+            setpointLimits.deadBand = kDefaultDeadBand;
+        }
+        setpointLimits.deadBandTemp = static_cast<int16_t>(setpointLimits.deadBand * 10);
+    }
+
+    if (AbsMinCoolSetpointLimit::Get(endpoint, &setpointLimits.absMinCoolSetpointLimit) != Status::Success)
+        setpointLimits.absMinCoolSetpointLimit = kDefaultAbsMinCoolSetpointLimit;
+
+    if (AbsMaxCoolSetpointLimit::Get(endpoint, &setpointLimits.absMaxCoolSetpointLimit) != Status::Success)
+        setpointLimits.absMaxCoolSetpointLimit = kDefaultAbsMaxCoolSetpointLimit;
+
+    if (MinCoolSetpointLimit::Get(endpoint, &setpointLimits.minCoolSetpointLimit) != Status::Success)
+        setpointLimits.minCoolSetpointLimit = setpointLimits.absMinCoolSetpointLimit;
+
+    if (MaxCoolSetpointLimit::Get(endpoint, &setpointLimits.maxCoolSetpointLimit) != Status::Success)
+        setpointLimits.maxCoolSetpointLimit = setpointLimits.absMaxCoolSetpointLimit;
+
+    if (AbsMinHeatSetpointLimit::Get(endpoint, &setpointLimits.absMinHeatSetpointLimit) != Status::Success)
+        setpointLimits.absMinHeatSetpointLimit = kDefaultAbsMinHeatSetpointLimit;
+
+    if (AbsMaxHeatSetpointLimit::Get(endpoint, &setpointLimits.absMaxHeatSetpointLimit) != Status::Success)
+        setpointLimits.absMaxHeatSetpointLimit = kDefaultAbsMaxHeatSetpointLimit;
+
+    if (MinHeatSetpointLimit::Get(endpoint, &setpointLimits.minHeatSetpointLimit) != Status::Success)
+        setpointLimits.minHeatSetpointLimit = setpointLimits.absMinHeatSetpointLimit;
+
+    if (MaxHeatSetpointLimit::Get(endpoint, &setpointLimits.maxHeatSetpointLimit) != Status::Success)
+        setpointLimits.maxHeatSetpointLimit = setpointLimits.absMaxHeatSetpointLimit;
+
+    if (setpointLimits.coolSupported)
+    {
+        if (OccupiedCoolingSetpoint::Get(endpoint, &setpointLimits.occupiedCoolingSetpoint) != Status::Success)
+        {
+            ChipLogError(Zcl, "Error: Can not read Occupied Cooling Setpoint");
+            return Status::Failure;
```

Comment:
Is there a reason we're not returning the actual status from the Get() here?  Similar in various places below.

If there is a reason (e.g. we don't want to claim UnsupportedAttribute for an attribute write to some attribute that's different from the one being actually gotten here), it would be good to document that reason.

Reply:
@hasty my point is that if it's important to not propagate the return value, that should be documented in the code, so someone does not "fix" it later.

Reply:
Added comments to the lines that do this.